timemachine/fe/topology.py

```diff
@@ -403,30 +403,27 @@ def setup_chiral_restraints(self, restraint_k=1000.0):
             Returns a ChiralAtomRestraint and a ChiralBondRestraint
 
         """
-        chiral_bonds = chiral_utils.find_chiral_bonds(self.mol)
+        mol = self.mol
+        conf = get_romol_conf(mol)
 
-        chiral_atom_restr_idxs = chiral_utils.setup_all_chiral_atom_restr_idxs(self.mol, get_romol_conf(self.mol))
-
-        chiral_atom_params = []
-        for idxs in chiral_atom_restr_idxs:
-            chiral_atom_params.extend(restraint_k for _ in idxs)
-
-        chiral_atom_params = np.array(chiral_atom_params)
-        chiral_atom_restr_idxs = np.array(chiral_atom_restr_idxs)
+        # chiral atoms
+        chiral_atom_restr_idxs = np.array(chiral_utils.setup_all_chiral_atom_restr_idxs(mol, conf))
+        chiral_atom_params = restraint_k * np.ones(len(chiral_atom_restr_idxs))
+        assert len(chiral_atom_params) == len(chiral_atom_restr_idxs)  # TODO: can this be checked in Potential::bind ?
```

Comment:
This is checked in `ChiralAtomRestraint::execute_device`, though might be a little too far from construction to be useful.